pgjdbc/src/test/java/org/postgresql/test/jdbc2/DateTest.java

```diff
@@ -46,114 +52,137 @@ public void tearDown() throws Exception {
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'2003-07-09'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'1934-02-28'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'1969-04-03'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'1982-08-03'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'2012-03-15'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'1912-05-01'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'1971-12-15'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'1984-12-03'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'2000-01-01'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'3456-01-01'")));
-    assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL("testdate", "'0101-01-01 BC'")));
-
-    /* dateTest() contains all of the tests */
-    dateTest();
-
-    assertEquals(18, stmt.executeUpdate("DELETE FROM " + "testdate"));
-    stmt.close();
+    for (String table : Arrays.asList("testdate", "testts", "testtstz")) {
+      Properties properties = new Properties();
+      try (Connection con = TestUtil.openDB(properties)) {
+        Statement stmt = con.createStatement();
+
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1950-02-07'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1970-06-02'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1999-08-11'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'2001-02-13'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1950-04-02'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1970-11-30'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1988-01-01'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'2003-07-09'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1934-02-28'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1969-04-03'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1982-08-03'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'2012-03-15'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1912-05-01'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1971-12-15'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'1984-12-03'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'2000-01-01'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'3456-01-01'")));
+        assertEquals(1, stmt.executeUpdate(TestUtil.insertSQL(table, "'0101-01-01 BC'")));
```

Comment:
This fails for me. 
expected:<0101-01-01> but was:<0102-12-31>

Reply:
My Postgres server has UTC+2 and it passes. So it looks like an issue with timezone of returned text string. I was expecting something like that.

I may need to write a similar test for the TimestampUtils parser, maybe I indeed broke this unintended.